packages/graphql/src/translate/create-set-relationship-properties.ts

```diff
@@ -85,6 +85,36 @@ function createSetRelationshipProperties({
             return;
         }
 
+        const arrayPushField = relationship.primitiveFields.find((x) => `${x.fieldName}_PUSH` === key);
+
+        if (arrayPushField) {
+            if (propertiesEntries.find(([entryKey]) => entryKey === arrayPushField.dbPropertyName)) {
+                throw new Error(`Ambiguous property: ${arrayPushField.dbPropertyName}`);
+            }
+            const pointArrayField = relationship.pointFields.find((x) => `${x.fieldName}_PUSH` === key);
+            if (pointArrayField) {
+                strs.push(
+                    `SET ${varName}.${arrayPushField.dbPropertyName} = ${varName}.${arrayPushField.dbPropertyName} + [p in $${paramName} | point(p)]`
+                );
+            } else {
+                strs.push(
+                    `SET ${varName}.${arrayPushField.dbPropertyName} = ${varName}.${arrayPushField.dbPropertyName} + $${paramName}`
+                );
+            }
+
+            return;
+        }
+
+        const arrayPopField = relationship.primitiveFields.find((x) => `${x.fieldName}_POP` === key);
+        if (arrayPopField) {
+            if (propertiesEntries.find(([entryKey]) => entryKey === arrayPopField.dbPropertyName)) {
+                throw new Error(`Ambiguous property: ${arrayPopField.dbPropertyName}`);
+            }
+            strs.push(
+                `SET ${varName}.${arrayPopField.dbPropertyName} = ${varName}.${arrayPopField.dbPropertyName}[0..-$${paramName}]`
+            );
+        }
```

Comment:
This seems to be a borderline duplication of the same code in `createUpdateAndParams` and I wonder if we could potentially do something to reduce duplication here.

Reply:
I agree the code looks very similar, but I think the results are distinct enough to warrant separate code paths. Do you have any concrete examples of deduplication I could use?

Reply:
Not particularly, it was more a gut feel! A lot of this will change in Cypher Builder land, so let's leave it as it is for now. 🙂